ema.c

```diff
@@ -318,6 +317,12 @@ void push_back_ema(ema_root_t *root, ema_t *node)
 int search_ema_range(ema_root_t *root, size_t start, size_t end,
                         ema_t **ema_begin, ema_t **ema_end)
 {
+    if ((start >= root->end) || (end <= root->start)) {
```

Comment:
Maybe move this check to caller. Only needed for sgx_mm_alloc, mm_alloc, mm_init_ema to check if requested range in user range or not.
Other cases, commit_x, modify_x, etc. should be covered by examine the nodes below.
RTS range may be non-continuous too.